lib/hanami/view/helpers/tag_helper.rb

```diff
@@ -0,0 +1,202 @@
+        #
+        # @example
+        #   link_to("Home", "/")
+        #   # => <a href="/">Home</a>
+        #
+        #   link_to("/") { "Home" }
+        #   # => <a href="/">Home</a>
+        #
+        #   link_to("Home", "/", class: "button") %>
+        #   # => <a href="/" class="button">Home</a>
+        #
+        # @example Escaping
+        #   link_to("<script>alert('xss')</script>", "/")
+        #   # => <a href="/">&lt;script&gt;alert(&#39;xss&#39;)&lt;/script&gt;</a>
+        #
+        #   link_to("/") { "<script>alert('xss')</script>" }
+        #   # => <a href="/">&lt;script&gt;alert(&#39;xss&#39;)&lt;/script&gt;</a>
+        #
+        # @see #tag
+        #
+        # @api public
+        # @since 2.0.0
+        def link_to(content, url = nil, **attributes, &block)
+          if block
+            raise ArgumentError if url && content
+
+            url = content
+            content = nil
+          end
+
+          attributes[:href] = url or raise ArgumentError
+
+          tag.a(content, **attributes, &block)
+        end
+
+        # Returns a string of space-separated tokens from a range of given arguments.
+        #
+        # This is intended for building an HTML tag attribute value, such as a list of class names.
+        #
+        # @return [String]
+        #
+        # @example
+        #   token_list("foo", "bar")
+        #   # => "foo bar"
+        #
+        #   token_list("foo", "foo bar")
+        #   # => "foo bar"
+        #
+        #   token_list({ foo: true, bar: false })
+        #   # => "foo"
+        #
+        #   token_list(nil, false, 123, "", "foo", { bar: true })
+        #   # => "123 foo bar"
+        #
+        # @api public
+        # @since 2.0.0
+        def token_list(*args)
+          tokens = build_tag_values(*args).flat_map { |value|
+            safe = value.html_safe?
+            value.split(/\s+/).map { |s| safe ? s.html_safe : s }
+          }
+
+          EscapeHelper.escape_join(tokens, " ")
+        end
+
+        # @see #token_list
+        #
+        # @api public
+        # @since 2.0.0
+        def class_names(...)
+          token_list(...)
+        end
+
+        # @api private
+        # @since 2.0.0
+        def build_tag_values(*args)
+          tag_values = []
+
+          args.each do |tag_value|
+            case tag_value
+            when Hash
+              tag_value.each do |key, val|
+                tag_values << key.to_s if val && !key.to_s.empty?
+              end
+            when Array
+              tag_values.concat build_tag_values(*tag_value)
+            else
+              tag_values << tag_value.to_s unless tag_value.to_s.empty?
+            end
+          end
+
+          tag_values
+        end
+
+        # @api private
+        # @since 2.0.0
+        def tag_builder
+          @tag_builder ||= begin
+            inflector = respond_to?(:_context) ? _context.inflector : Dry::Inflector.new
```

Comment:
@timriley this is missing a `require "dry/inflector"` for this ternary operator's "else" case.

I'm testing this git branch while developing Assets:

```shell
Failures:

  1) Hanami::Assets::Helpers#js returns the path to the asset
     Failure/Error: tag.script(**attributes).to_s

     NameError:
       uninitialized constant Hanami::View::Helpers::TagHelper::Dry
     # /Users/jodosha/.gem/ruby/3.2.2/bundler/gems/view-38777a3ef52b/lib/hanami/view/helpers/tag_helper.rb:195:in `tag_builder'
     # /Users/jodosha/.gem/ruby/3.2.2/bundler/gems/view-38777a3ef52b/lib/hanami/view/helpers/tag_helper.rb:73:in `tag'
     # ./lib/hanami/assets/helpers.rb:206:in `block in javascript'
     # ./lib/hanami/assets/helpers.rb:855:in `map'
     # ./lib/hanami/assets/helpers.rb:855:in `_safe_tags'
     # ./lib/hanami/assets/helpers.rb:194:in `javascript'
     # ./spec/unit/hanami/assets/helpers_spec.rb:89:in `block (3 levels) in <top (required)>'
```

Reply:
@jodosha Nice find! Sorted this now in https://github.com/hanami/view/pull/229/commits/a05ea7542bc07571411af793823781980488f454.